### ISSUE TITLE
select on stdin/stdout

### ISSUE THREAD

**Issue by SilvanScherrer** [bitwiseworks/libcx#84]
It would be very good to have select also working on stdin/stdout. stdout is easy, as it just is writable always. stdin need probably some KbdPeek() scheme to see if some keystrokes are available. and the set the fd according.

**Comment by dmik** [comment]
Yes, it definitely makes sense, I've just fixed some related stuff, see https://github.com/bitwiseworks/openssh-portable/commit/e46de058faede989d5fe0be3fb8e1d941dc3d944 and https://github.com/bitwiseworks/openssh-portable/issues/5.

**Comment by dmik** [comment]
There is an implementation with similar functionality from @komh https://github.com/komh/os2compat/blob/master/network/select.c.

We need something like that. Current openssh solution and many other similar fixes are polling based, which is very limited. Semaphores and named pipes is the right approach.

I'll postpone it for the next cycle as it's not trivial.
